File: it/structures/csharp/tuple-default/Test/Entry.cs
using Newtonsoft.Json;
using System;
using System.Text;

namespace Test {
    [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
    public class Entry {
        [JsonProperty("tuple1")]
        public Tuple1 tuple1 {
            get;
        }

        [JsonProperty("tuple2")]
        public Tuple2 tuple2 {
            get;
        }

        [JsonConstructor]
        public Entry (
            [JsonProperty("tuple1")] Tuple1 tuple1,
            [JsonProperty("tuple2")] Tuple2 tuple2
        ) {
            this.tuple1 = tuple1;
            this.tuple2 = tuple2;
        }

        public override bool Equals(Object other) {
            Entry o = other as Entry;

            if (o == null) {
                return false;
            }

            if (this.tuple1 == null) {
                if (o.tuple1 != null) {
                    return false;
                }
            } else {
                if (!this.tuple1.Equals(o.tuple1)) {
                    return false;
                }
            }

            if (this.tuple2 == null) {
                if (o.tuple2 != null) {
                    return false;
                }
            } else {
                if (!this.tuple2.Equals(o.tuple2)) {
                    return false;
                }
            }

            return true;
        }

        public override int GetHashCode() {
            int result = 1;
            result = result * 31 + this.tuple1.GetHashCode();
            result = result * 31 + this.tuple2.GetHashCode();
            return result;
        }

        public override String ToString() {
            StringBuilder b = new StringBuilder();

            b.Append("Entry(");
            b.Append("tuple1=");
            b.Append(this.tuple1);
            b.Append(", ");
            b.Append("tuple2=");
            b.Append(this.tuple2);
            b.Append(")");

            return b.ToString();
        }
    }
}

File: it/structures/java/tuple-default/test/Entry.java
package test;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.Optional;

public class Entry {
    @JsonProperty("tuple1")
    final Optional<Tuple1> tuple1;
    @JsonProperty("tuple2")
    final Optional<Tuple2> tuple2;

    @JsonCreator
    public Entry(
        @JsonProperty("tuple1") Optional<Tuple1> tuple1,
        @JsonProperty("tuple2") Optional<Tuple2> tuple2
    ) {
        this.tuple1 = tuple1;
        this.tuple2 = tuple2;
    }

    @JsonProperty("tuple1")
    public Optional<Tuple1> getTuple1() {
        return this.tuple1;
    }

    @JsonProperty("tuple2")
    public Optional<Tuple2> getTuple2() {
        return this.tuple2;
    }

    @Override
    public String toString() {
        final StringBuilder b = new StringBuilder();

        b.append("Entry(");
        b.append("tuple1=");
        b.append(this.tuple1.toString());
        b.append(", ");
        b.append("tuple2=");
        b.append(this.tuple2.toString());
        b.append(")");

        return b.toString();
    }

    @Override
    public int hashCode() {
        int result = 1;
        final StringBuilder b = new StringBuilder();
        result = result * 31 + this.tuple1.hashCode();
        result = result * 31 + this.tuple2.hashCode();
        return result;
    }

    @Override
    public boolean equals(final Object other_) {
        if (other_ == null) {
            return false;
        }

        if (!(other_ instanceof Entry)) {
            return false;
        }

        @SuppressWarnings("unchecked")
        final Entry o_ = (Entry)other_;

        if (!this.tuple1.equals(o_.tuple1)) {
            return false;
        }

        if (!this.tuple2.equals(o_.tuple2)) {
            return false;
        }

        return true;
    }

    public static class Builder {
        private Optional<Tuple1> tuple1;
        private Optional<Tuple2> tuple2;

        private Builder() {
            this.tuple1 = Optional.empty();
            this.tuple2 = Optional.empty();
        }

        public Entry build() {

            return new Entry(
                this.tuple1,
                this.tuple2
            );
        }

        public Builder tuple1(final Tuple1 tuple1) {
            this.tuple1 = Optional.of(tuple1);
            return this;
        }

        public Builder tuple2(final Tuple2 tuple2) {
            this.tuple2 = Optional.of(tuple2);
            return this;
        }
    }

    /**
     * Construct a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }
}
